pyproject.toml

```diff
@@ -579,6 +579,8 @@ exclude = [
 ignore = [
   "ANN101", # Missing type annotation for `self` in method
   "ANN102", # Missing type annotation for `cls` in class method
+  "EM101",  # Exception must not use a string literal, assign to a variable first
+  "EM102",  # Exception must not use an f-string literal, assign to a variable first
```

Comment:
This would mean we're ignoring 2/3 EM rules 😅.

If too many failures are triggered by enabling EM, you could always do something like

```
ruff check --select=EM --add-noqa src tests
```

and that would allow us to adopt this rule incrementally.

Reply:
Thanks for bringing it up. Adding a `noqa` seems like a good starting point for now. I have removed `EM101` and `EM102` from ignore list.